pkg/provisioner/templates/nv-driver.go

```diff
@@ -34,21 +34,27 @@ wget https://developer.download.nvidia.com/compute/cuda/repos/$distribution/x86_
 sudo dpkg -i cuda-keyring_1.1-1_all.deb
 
 with_retry 3 10s sudo apt-get update
-install_packages_with_retry cuda-drivers
+install_packages_with_retry cuda-drivers{{if .Version}}={{.Version}}{{end}}
 
 nvidia-smi -L
 `
 
 type NvDriver struct {
+	Version string
 }
 
-func NewNvDriver() *NvDriver {
-	return &NvDriver{}
+func NewNvDriver(env v1alpha1.Environment) *NvDriver {
+	if env.Spec.NVIDIADriver.Version == "" {
+		return &NvDriver{}
+	}
+	return &NvDriver{
+		Version: env.Spec.NVIDIADriver.Version,
+	}
```

Comment:
```suggestion
	return &NvDriver{
		Version: env.Spec.NVIDIADriver.Version,
	}
```
The check for the empty string is redundant since golang uses `""` as a zero value for strings anyway.